### ISSUE TITLE
When will java 16 be supported?

### ISSUE THREAD

**Issue by 030** [libgdx/libgdx#6613]
Gretty issue seems to be resolved https://github.com/gretty-gradle-plugin/gretty/issues/231#issuecomment-884688985

**Comment by NathanSweet** [comment]
What isn't supported with Java 16?

**Comment by 030** [comment]
https://libgdx.com/dev/setup/#2-idea

> At the moment, libGDX projects do not work with JDK 16, as Gretty does not yet support Gradle 7. As a consequence, you are advised to use JDK 8-15!

My system:

```
$ java --version
openjdk 16.0.1 2021-04-20
OpenJDK Runtime Environment AdoptOpenJDK-16.0.1+9 (build 16.0.1+9)
OpenJDK 64-Bit Server VM AdoptOpenJDK-16.0.1+9 (build 16.0.1+9, mixed mode, sharing)
```
gradle:

```
./gradlew build
> Task :android:compileDebugRenderscript FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Some problems were found with the configuration of task ':android:compileDebugRenderscript' (type 'RenderscriptCompile').
  - Type 'com.android.build.gradle.tasks.RenderscriptCompile' method 'useAndroidX()' should not be annotated with: @Input.
    
    Reason: Input/Output annotations are ignored if they are placed on something else than a getter.
    
    Possible solutions:
      1. Remove the annotations.
      2. Rename the method.
    
    Please refer to https://docs.gradle.org/7.0.2/userguide/validation_problems.html#ignored_annotations_on_method for more details about this problem.
  - Type 'com.android.build.gradle.tasks.RenderscriptCompile' property 'ndkConfig' is missing an input or output annotation.
    
    Reason: A property without annotation isn't considered during up-to-date checking.
    
    Possible solutions:
      1. Add an input or output annotation.
      2. Mark it as @Internal.
    
    Please refer to https://docs.gradle.org/7.0.2/userguide/validation_problems.html#missing_annotation for more details about this problem.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/7.0.2/userguide/command_line_interface.html#sec:command_line_warnings
```

**Comment by tommyettinger** [comment]
You can use [gdx-liftoff](https://github.com/tommyettinger/gdx-liftoff) to make a project, which defaults to Gradle 7.x and as such makes projects that are compatible with Java 16. Note, the Android build tools from Google are not yet compatible with Java 16, so you may want to have Java 8 or 11 around anyway. Since libGDX is actually compatible with Java 16 (just gdx-setup isn't), maybe this can be closed. There's a workaround used by gdx-liftoff to get the changes that have been released in 3.0.6, and that's been used since... maybe April. I might be able to change that to just depend on Gretty 3.0.6 now.

**Comment by 030** [comment]
Thank you. I can now run a sample.
